src/init.cpp

```diff
@@ -1423,10 +1423,12 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
                         CleanupBlockRevFiles();
                 }
 
+                LogPrintf("Loading block index...");
                 if (!LoadBlockIndex()) {
                     strLoadError = _("Error loading block database");
                     break;
                 }
+                if (ShutdownRequested()) break;
```

Comment:
This check is needed because if shutdown is requested while in `LoadBlockIndex()`, it will return `true` because the arrival of a shutdown request is not an error.